wade-review.md

```diff
@@ -0,0 +1,596 @@
+
+
+```java
+
+    public static void main(final String[] args) throws Exception {
+
+        final Properties props = GitHubPrRatio.loadEnvProperties("get-started.properties");
+
+        props.put(StreamsConfig.APPLICATION_ID_CONFIG, "streams-github-87");
+        props.put(StreamsConfig.STATESTORE_CACHE_MAX_BYTES_CONFIG, 0);
+        props.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+
+        GitHubPrRatio gitHubPrRatio = new GitHubPrRatio();
+
+
+}
+
+I’ll call the  GitHubPrRatio method, to get the ratio coming in.
+
+
+```java
+
+    public static void main(final String[] args) throws Exception {
+
+        final Properties props = GitHubPrRatio.loadEnvProperties("get-started.properties");
+
+        props.put(StreamsConfig.APPLICATION_ID_CONFIG, "streams-github-87");
+        props.put(StreamsConfig.STATESTORE_CACHE_MAX_BYTES_CONFIG, 0);
+        props.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+
+        GitHubPrRatio gitHubPrRatio = new GitHubPrRatio();
+
+        final KafkaStreams streams = new KafkaStreams(gitHubPrRatio.topology(props), props);
+
+}
+
+Create a new Streams instance with our topology. What’s a topology? Hold on a sec, I’ll show you, but first, let’s tie this up.
+
+
+```java
+
+    public static void main(final String[] args) throws Exception {
+
+        final Properties props = GitHubPrRatio.loadEnvProperties("get-started.properties");
+
+        props.put(StreamsConfig.APPLICATION_ID_CONFIG, "streams-github-87");
+        props.put(StreamsConfig.STATESTORE_CACHE_MAX_BYTES_CONFIG, 0);
+        props.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+
+        GitHubPrRatio gitHubPrRatio = new GitHubPrRatio();
+
+        final KafkaStreams streams = new KafkaStreams(gitHubPrRatio.topology(props), props);
+
+        final CountDownLatch latch = new CountDownLatch(1);
+        Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+            streams.close(Duration.ofSeconds(5));
+            latch.countDown();
+        }));
+}
+
+Then we’ll set up a countdown latch so it waits for other threads before it starts…
+
+
+```java
+
+    public static void main(final String[] args) throws Exception {
+
+        final Properties props = GitHubPrRatio.loadEnvProperties("get-started.properties");
+
+        props.put(StreamsConfig.APPLICATION_ID_CONFIG, "streams-github-87");
+        props.put(StreamsConfig.STATESTORE_CACHE_MAX_BYTES_CONFIG, 0);
+        props.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+
+        GitHubPrRatio gitHubPrRatio = new GitHubPrRatio();
+
+        final KafkaStreams streams = new KafkaStreams(gitHubPrRatio.topology(props), props);
+
+        final CountDownLatch latch = new CountDownLatch(1);
+        Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+            streams.close(Duration.ofSeconds(5));
+            latch.countDown();
+        }));
+
+        try {
+            streams.start();
+            latch.await();
```

Comment:
Oh, I guess this is where you wait for the latch?

Reply:
https://github.com/Cerchie/notes-for-gh-video/pull/2/commits/dadfe630441c7a138602ba697f7ddec154d4b10f